magicblock-committor-service/src/intent_executor/intent_executor.rs

```diff
@@ -0,0 +1,247 @@
+    ) -> IntentExecutorResult<Signature, (InternalError, Option<Signature>)>
+    {
+        let latest_blockhash = self
+            .rpc_client
+            .get_latest_blockhash()
+            .await
+            .map_err(|err| (err.into(), None))?;
+        match &mut prepared_message {
+            VersionedMessage::V0(value) => {
+                value.recent_blockhash = latest_blockhash;
+            }
+            VersionedMessage::Legacy(value) => {
+                warn!("TransactionPreparator v1 does not use Legacy message");
+                value.recent_blockhash = latest_blockhash;
+            }
+        };
+
+        let transaction =
+            VersionedTransaction::try_new(prepared_message, &[&self.authority])
+                .map_err(|err| (err.into(), None))?;
+        let result = self
+            .rpc_client
+            .send_transaction(
+                &transaction,
+                &MagicBlockSendTransactionConfig::ensure_committed(),
+            )
+            .await
+            .map_err(|err| {
+                let signature = err.signature();
+                (err.into(), signature)
+            })?;
+
+        Ok(result.into_signature())
+    }
+
+    fn persist_result<P: IntentPersister>(
+        persistor: &Option<P>,
+        result: &IntentExecutorResult<ExecutionOutput>,
+        message_id: u64,
+        pubkeys: &[Pubkey],
+    ) {
+        match result {
+            Ok(value) => {
+                let signatures = CommitStatusSignatures {
+                    process_signature: value.commit_signature,
+                    finalize_signature: Some(value.commit_signature)
+                };
+                let update_status = CommitStatus::Succeeded(signatures);
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::EmptyIntentError) => {
+                let update_status = CommitStatus::Failed;
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::FailedToFitError)) => {
+                let update_status = CommitStatus::PartOfTooLargeBundleToProcess;
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::TaskBuilderError(err))) => {
+                match err {
+                    crate::tasks::task_builder::Error::CommitTasksBuildError(_) => {
+                        let update_status = CommitStatus::Failed;
+                        persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+                    }
+                    crate::tasks::task_builder::Error::FinalizedTasksBuildError(_) => {
+                        // TODO: commit signature to set this
```

Comment:
Needs to be addressed before merging no?
Otherwise we just end up ignoring that error.

Reply:
Ah, This TODO has to be removed, since there's no "Finalize" tasks involved in commit flow